manifests/init.pp

```diff
@@ -90,4 +92,8 @@
     pattern   => '/usr/sbin/in.tftpd',
     start     => $start,
   }
+
+  if $files {
```

Comment:
+1 to this pattern. All defines should be able to be expressed this way because it allows you to specify hashes in Hiera for that data.

Would suggest that you default to using hiera_hash() so tftp::file can be expressed at multiple levels in Hiera's hierarchy.

Here's an example

https://github.com/ghoneycutt/puppet-module-types/blob/master/manifests/init.pp